test/check-application

```diff
@@ -2139,7 +2139,7 @@ class TestFiles(testlib.MachineCase):
         b.click("#copy-item")
         b.go("files#/?path=/home/foouser")
         self.assert_last_breadcrumb("foouser")
-        b.wait_visible("[data-item='foouserfile']")
+        b.wait_not_present(".pf-v5-c-empty-state")
```

Comment:
This now smells like a race -- waiting for a negative immediately after navigating a page will be pretty much always true. You first need to wait for something "positive" on the page, i.e. what is rendered instead of the empty state. Or first wait for the empty state to appear (maybe by moving into a new empty directory)?

Reply:
The previous problem was a race as `foouserfile` exists in /home/foouser and in /home/admin, but waiting on the empty state to go away is not a race as we navigate from `/home/admin` to `/home/foouser` and that first shows an empty state and then files.

The issue in all these cases is that it is hard to assert when `fsinfo` is done as `assert_last_breadcrumb` only checks the `path` state but not if `fsinfo` has loaded the directory.  So in this case we could instead check the owner of the directory that comes from fsinfo and what I did in 8f28589e7cd515b2a3d740544dd77452c0fbca5f. That should go from `foouser` to `admin`.

Reply:
The pattern seems obvious?
 - change from empty to non-empty dir: wait for empty state to go away
 - change from non-empty to empty: wait for empty state to appear
 - change from non-empty to non-empty: wait for "old" file to go away and "new" file to appear

This is the third case, so empty state isn't a steady state (perhaps a brief flash during initialization). So couldn't this wait for a known file in /home/foouser/ to appear instead?

Reply:
Nope, you can't wait on a different file as both dirs have the same files but we can wait on the bottom bar to show a different owner.

Reply:
Ah, so add (on the CLI) a new file into /home/foouser/ for which you can wait? Or show the owner change, yes.

Reply:
Did the owner thing, so I can focus on getting https://github.com/cockpit-project/cockpit-files/pull/953 green